### ISSUE TITLE
SSL support not working in strapi v3, we need it for facebook authentication

### ISSUE THREAD

**Issue by manojkumar7488** [strapi/strapi#1102]
SSL support not working in strapi v3
we can try :
1.
![screen shot 2018-05-03 at 7 30 25 pm](https://user-images.githubusercontent.com/13350196/39581699-d0d502f6-4f09-11e8-9677-66441568c1b1.png)

but in this way render only  https://site_urlindex.html

please give me suggession how to do facebook authentication using strapi framework locally

**Comment by soupette** [comment]
Hi @manojkumar7488 maybe you can try to follow this [tutorial](https://github.com/strapi/strapi-examples/blob/master/login-react/doc/github_setup.md) but with Facebook

**Comment by manojkumar7488** [comment]
Hello @soupette  after login i got 

{"statusCode":400,"error":"Bad Request","message":"This provider is disabled."}

Why this error

**Comment by soupette** [comment]
Did you enable it in the `users-permissions` plugin?

**Comment by manojkumar7488** [comment]
yes i enable it

**Comment by jrodriguesimg** [comment]
I also have problems with the facebook provider. The redirect_uri sent to facebook uses http instead of the obligatory https. Should I create a different issue?

**Comment by gianniskarmas** [comment]
Why do you just keep closing issues?
The problems are still there unfixed and create so much frustration.

I'm also trying to enable ssl in strapi and followed [this](https://github.com/strapi/strapi/issues/171#issuecomment-384679495) but it gives the wrong redirect of "https://site_urlindex.html".

In nuxt 2.0 they very nicely implemented a simple solution to enable ssl in nuxt.config.
This is just brilliant and makes developing so much easier.
```
    server: {
        port: 443, // default: 3000
        host: '0.0.0.0', // default: localhost,
        https: {
            key: fs.readFileSync(path.resolve(__dirname, 'key.key')),
            cert: fs.readFileSync(path.resolve(__dirname, 'cert.pem'))
        }
    },
```
WE need to have the choice of when to use nginx or any other tool to do the job.
ngrok is great but not a solution as you need to send everything to the net and back.

IMHO the choice of removing it was wrong and It is mandatory to be able to enable easily ssl in strapi.

Having said that, can someone please post a code sample of what we have to do to enable ssl support in strapi with a simple node command and not having to setup nginx?

Many Thanks

**Comment by derrickmehaffy** [comment]
@gianniskarmas all SSL native support was removed from Strapi and isn't supported here, you will need to proxy it. As for the https URI for providers, the issue is known and is being worked on.

Using a 3rd party software for proxying SSL for node.js based applications is not new and is typically the industry standard when you are working with nodejs (A google search will show you that).

If you wanted to try and build in proper support for SSL then be my guest, submit a PR and prove how easy it is, but the official stance (AFAIK in my talks with the Strapi team) it will not be supported naively at any point in the near future.

Nginx is simple and easy to setup, even if you can't run it on the same system that Strapi is running on. If you need help setting it up please ping me the on the slack and I can give you a little guide, else you can read my WIP Medium article here: https://medium.com/@derrickmehaffy/using-strapi-in-production-with-https-717af82d1445 (Might be a few typos as I haven't had the time to proof read it)

**Comment by synth3tk** [comment]
@gianniskarmas They mentioned (at least in this issue) that the issue creation template must be followed, or else they'll close the ticket until it's updated to reflect the template. It's not that they're saying it's an invalid or fixed issue.

**Comment by derrickmehaffy** [comment]
Yes that is the case, if you do not follow the template your issue/FR is closed with absolutely no additional troubleshooting done, the same applies if your issue is a duplicate.

See the below snippet of the issue template:

```
<!-- ⚠️ If you do not respect this template your issue will be closed. -->

<!-- =============================================================================== -->
<!-- ⚠️ If you are not using the current Strapi release, you will be asked to update. -->
<!-- Please see the wiki for guides on upgrading to the latest release. -->
<!-- =============================================================================== -->

<!-- ⚠️ Make sure to browse the opened and closed issues before submitting your issue. -->

<!-- ⚠️ Before writing your issue make sure you are using:-->
<!-- Node 10.x.x -->
<!-- npm 6.x.x -->
<!-- The latest version of Strapi. -->

**Informations**
- **Node.js version**: 
- **npm version**: 
- **Strapi version**: <!-- Please make sure you are on the latest version -->
- **Database**: 
- **Operating system**: 
```

**Comment by emahuni** [comment]
See this: #4884
